diff --git a/index.js b/index.js
index e71e6d19efc5b6cf21d0e07f0de79feb4ee6d9f4..76c100adca305145fa79b77e9452bc60ece4bbba 100644
--- a/index.js
+++ b/index.js
@@ -1,4 +1,3 @@
-var bufferEqual = require('buffer-equal-constant-time');
 var Buffer = require('safe-buffer').Buffer;
 var crypto = require('crypto');
 var formatEcdsa = require('ecdsa-sig-formatter');
@@ -15,6 +14,14 @@ if (supportsKeyObjects) {
   MSG_INVALID_SECRET += 'or a KeyObject';
 }
 
+function timingSafeEqual(a, b) {
+  if (a.byteLength !== b.byteLength) {
+    return false;
+  }
+
+  return crypto.timingSafeEqual(a, b);
+}
+
 function checkIsPublicKey(key) {
   if (Buffer.isBuffer(key)) {
     return;
@@ -138,7 +145,7 @@ function createHmacSigner(bits) {
 function createHmacVerifier(bits) {
   return function verify(thing, signature, secret) {
     var computedSig = createHmacSigner(bits)(thing, secret);
-    return bufferEqual(Buffer.from(signature), Buffer.from(computedSig));
+    return timingSafeEqual(Buffer.from(signature), Buffer.from(computedSig));
   }
 }
 
diff --git a/package.json b/package.json
index 0777d53392d01fec7493b6f9ccccb3d431889cab..e2812ea51dabb3f90e2ba4cce3c53212e2fec2f5 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,6 @@
     "test": "test"
   },
   "dependencies": {
-    "buffer-equal-constant-time": "1.0.1",
     "ecdsa-sig-formatter": "1.0.11",
     "safe-buffer": "^5.0.1"
   },
